File: sesion-7/queries.py
from db_connection import create_connection, close_connection

def fetch_all_employees():
    """Obtener todos los empleados"""
    connection = create_connection()
    employees = []
    if connection:
        cursor = connection.cursor(dictionary=True)
        cursor.execute("SELECT * FROM employees")
        employees = cursor.fetchall()
        cursor.close()
        close_connection(connection)
    return employees

def create_employee(first_name, last_name, email, hire_date, salary):
    """Crear un nuevo empleado"""
    connection = create_connection()
    if connection:
        cursor = connection.cursor()
        query = """
        INSERT INTO employees (first_name, last_name, email, hire_date, salary)
        VALUES (%s, %s, %s, %s, %s)
        """
        cursor.execute(query, (first_name, last_name, email, hire_date, salary))
        connection.commit()
        cursor.close()
        close_connection(connection)
        print("Empleado creado exitosamente")

def update_employee(employee_id, first_name, last_name, email, salary):
    """Actualizar un empleado"""
    connection = create_connection()
    if connection:
        cursor = connection.cursor()
        query = """
        UPDATE employees
        SET first_name = %s, last_name = %s, email = %s, salary = %s
        WHERE id = %s
        """
        cursor.execute(query, (first_name, last_name, email, salary, employee_id))
        connection.commit()
        cursor.close()
        close_connection(connection)
        print("Empleado actualizado exitosamente")

def delete_employee(employee_id):
    """Eliminar un empleado"""
    connection = create_connection()
    if connection:
        cursor = connection.cursor()
        query = "DELETE FROM employees WHERE id = %s"
        cursor.execute(query, (employee_id,))
        connection.commit()
        cursor.close()
        close_connection(connection)
        print("Empleado eliminado exitosamente")

# Ejemplo de uso
if __name__ == "__main__":
    # Obtener todos los empleados
    employees = fetch_all_employees()
    print("Empleados:", employees)
    
    # Crear un nuevo empleado
    create_employee("Carlos", "Hernandez", "carlos.hernandez@example.com", "2023-01-01", 55000)
    
    # Actualizar un empleado existente
    update_employee(1, "Juan", "Perez", "juan.perez@example.com", 60000)
    
    # Eliminar un empleado
    delete_employee(2)

File: session-5/queries.js
const express = require("express");
const router = express.Router();
const db = require("./db");

// Obtener todos los empleados
router.get("/employees", (req, res) => {
  const query = "SELECT * FROM employees";
  db.query(query, (err, results) => {
    if (err) {
      console.error("Error fetching employees: " + err.stack);
      res.status(500).send("Error fetching employees");
      return;
    }
    res.json(results);
  });
});

// Crear un nuevo empleado
router.post("/employees", (req, res) => {
  const { first_name, last_name, email, hire_date, salary } = req.body;
  const query =
    "INSERT INTO employees (first_name, last_name, email, hire_date, salary) VALUES (?, ?, ?, ?, ?)";
  db.query(
    query,
    [first_name, last_name, email, hire_date, salary],
    (err, results) => {
      if (err) {
        console.error("Error creating employee: " + err.stack);
        res.status(500).send("Error creating employee");
        return;
      }
      res.status(201).json({ id: results.insertId });
    }
  );
});

// Actualizar un empleado
router.put("/employees/:id", (req, res) => {
  const { id } = req.params;
  const { first_name, last_name, email, hire_date, salary } = req.body;
  const query =
    "UPDATE employees SET first_name = ?, last_name = ?, email = ?, hire_date = ?, salary = ? WHERE id = ?";
  db.query(
    query,
    [first_name, last_name, email, hire_date, salary, id],
    (err, results) => {
      if (err) {
        console.error("Error updating employee: " + err.stack);
        res.status(500).send("Error updating employee");
        return;
      }
      res.status(200).send("Employee updated");
    }
  );
});

// Eliminar un empleado
router.delete("/employees/:id", (req, res) => {
  const { id } = req.params;
  const query = "DELETE FROM employees WHERE id = ?";
  db.query(query, [id], (err, results) => {
    if (err) {
      console.error("Error deleting employee: " + err.stack);
      res.status(500).send("Error deleting employee");
      return;
    }
    res.status(200).send("Employee deleted");
  });
});

module.exports = router;
